simulator/modules/fetch/bpu/t/unit_test.cpp

```diff
@@ -15,27 +15,29 @@ TEST_CASE( "Initialization: WrongParameters")
     CHECK_THROWS_AS( BaseBP::create_bp( "saturating_two_bits", 100, 20), BPInvalidMode);
 }
 
-//TEST_CASE( "Static, all branches not taken")
-//{
-//    auto bp = BaseBP::create_bp( "always_not_taken", 128, 16);
-//
-//    Addr PC = 28;
-//    Addr target = 12;
-//    
-//    CHECK_FALSE( bp->is_taken(PC) );
-//    CHECK( bp->get_target(PC) == target);
-//}
-
-//TEST_CASE( "Static, all branches taken")
-//{
-//    auto bp = BaseBP::create_bp( "always_taken", 128, 16);
-//
-//    Addr PC = 28;
-//    Addr target = 12;
-//    
-//    CHECK( bp->is_taken(PC) );
-//    CHECK( bp->get_target(PC) == target);
-//}
+TEST_CASE( "Static, all branches not taken")
+{
+    auto bp = BaseBP::create_bp( "always_not_taken", 128, 16);
+
+    Addr PC = 28;
+    Addr target = 12;
+
+    bp->update( BPInterface( PC, false, target));
+    CHECK_FALSE( bp->is_taken(PC) );
+    //CHECK( bp->get_target(PC) == target);
```

Comment:
now only this check fails with 32 == 12

Reply:
That's right. If branch is not taken, its should be PC + 4, i.e. 32

Reply:
So, I will change CHECK to CHECK_FALSE, I got it

Reply:
I'd leave the stronger assertion of CHECK( bp->get_target( PC) == PC + 4);

Reply:
Done